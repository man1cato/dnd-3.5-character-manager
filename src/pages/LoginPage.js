import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../store/actions/auth';

export const LoginPage = ({startLogin}) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">D&D 3.5e Character Manager</h1>
            <button className="button--primary" onClick={startLogin}>Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
