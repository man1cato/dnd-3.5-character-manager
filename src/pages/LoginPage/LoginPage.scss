@import '../../styles/base/settings';


.LoginPage {
   background: url('/images/bg.jpg');
   background-size: cover;
   background-position: right;
   display: flex;
   justify-content: center;
   align-items: center;
   height: 100vh;
   width: 100vw;       
}

.LoginPage__box {
   background: fade-out(white, .1);
   border-radius: 3px;
   text-align: center;
   width: 80vw;
   max-width: 50rem;
   min-width: 30rem;
   padding: $l-size;
}

.LoginPage__title {
   font-family: 'Roboto Slab', Roboto, Open-Sans, sans-serif;
   color: $dark-blue;
   line-height: 1;
   margin: 0 0 $l-size 0;
}