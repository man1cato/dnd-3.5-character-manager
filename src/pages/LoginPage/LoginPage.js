import React from 'react'
// import { connect } from 'react-redux'
import StyledFirebaseAuth from 'react-firebaseui/StyledFirebaseAuth'

import { firebase } from '../../firebase/firebase'
// import { startLogin } from '../../store/actions/auth'
import './LoginPage.scss'


const uiConfig = {
    signInFlow: 'popup',
    signInSuccessUrl: '/',
    signInOptions: [
        firebase.auth.GoogleAuthProvider.PROVIDER_ID,
        // firebase.auth.FacebookAuthProvider.PROVIDER_ID,
        // firebase.auth.TwitterAuthProvider.PROVIDER_ID,
        firebase.auth.EmailAuthProvider.PROVIDER_ID,
        // firebase.auth.PhoneAuthProvider.PROVIDER_ID
    ]
}

export const LoginPage = props => {
    return (
        <div className="LoginPage">
            <div className="LoginPage__box">
                <h1 className="LoginPage__title">D&D 3.5e Character Manager</h1>
                <StyledFirebaseAuth uiConfig={uiConfig} firebaseAuth={firebase.auth()} />
            </div>
        </div>
    )
}

// const mapDispatchToProps = dispatch => ({
//     startLogin: () => dispatch(startLogin())
// })

// export default connect(undefined, mapDispatchToProps)(LoginPage)

export default LoginPage
