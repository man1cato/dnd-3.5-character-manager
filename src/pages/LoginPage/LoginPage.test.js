import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import LoginPage from './LoginPage'


test('should render login page correctly', () => {
    const { container } = render(<LoginPage />)
    expect(container).toMatchSnapshot()
})

// test('should trigger startLogin on button click', () => {
//     const startLogin = jest.fn()
//     const wrapper = shallow(<LoginPage startLogin={startLogin} />)
//     wrapper.find('button').simulate('click')
//     expect(startLogin).toHaveBeenCalled()
// })