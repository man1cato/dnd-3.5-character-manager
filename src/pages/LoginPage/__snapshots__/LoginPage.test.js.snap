// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page correctly 1`] = `
<div
  className="LoginPage"
>
  <div
    className="LoginPage__box"
  >
    <h1
      className="LoginPage__title"
    >
      D&D 3.5e Character Manager
    </h1>
    <e
      firebaseAuth={
        Object {
          "apiKey": "AIzaSyChIDLvv9RpD9-kzAs5Z_r20fv3L0jYHjA",
          "appName": "[DEFAULT]",
          "authDomain": "dnd-3-5-character-manager-test.firebaseapp.com",
          "currentUser": null,
        }
      }
      uiConfig={
        Object {
          "signInFlow": "popup",
          "signInOptions": Array [
            "google.com",
            "password",
          ],
          "signInSuccessUrl": "/",
        }
      }
    />
  </div>
</div>
`;
